src/char/int_storage.c

```diff
@@ -41,24 +41,25 @@ struct inter_storage_interface inter_storage_s;
 struct inter_storage_interface *inter_storage;
 
 /// Save storage data to sql
-int inter_storage_tosql(int account_id, struct storage_data* p)
+int inter_storage_tosql(int account_id, const struct item *p)
```

Comment:
Ah correct, this should use and send the length of the vector through to `char_ memitemdata_to_sql`